include/mxnet/lib_api.h

```diff
@@ -1111,20 +1246,60 @@ extern "C" {
 
     // create a vector of tensors for inputs
     std::vector<MXTensor> inputs(num_in);
+    // create a vector for sparse inputs
+    std::vector<MXSparse> in_sparse(num_in);
```

Comment:
I think it will save the memory space to use `std::vector<MXTensor*> and std::vector<MXSparse*>`, or a union which contains the pointers of `MXTensor` and `Sparse`. However, it is a small thing : )

Reply:
how will this save memory?

I think it needs to be a vector<MXTensor> since this is where we're actually storing the tensors. when the function exits the memory is freed since this is on the stack.

Reply:
I see. `MXSparse` is the data of `MXTensor`. Sorry for my mistake.